src/deviceinfo.cpp

```diff
@@ -205,6 +217,52 @@ void DeviceInfoPrivate::modemSerialChanged(const QString &serial)
     updateModemPropertiesLater();
 }
 
+QString DeviceInfoPrivate::wlanMacAddress()
+{
+    return macAddress(DeviceInfoPrivate::WlanMode, 0);
+}
+
+int DeviceInfoPrivate::networkInterfaceCount(DeviceInfoPrivate::NetworkMode mode)
+{
+    /* Like QNetworkInfo::networkInterfaceCount() */
+    return networkModeDirectoryList(mode).size();
+}
+
+QString DeviceInfoPrivate::macAddress(DeviceInfoPrivate::NetworkMode mode, int interface)
+{
+    /* Like QNetworkInfo::macAddress() */
+    if (interface >= 0 && interface < networkInterfaceCount(mode))
+        return readSimpleFile(QDir(networkModeDirectoryList(mode).at(interface)).filePath("address"));
+    return QString();
+}
+
+const QStringList& DeviceInfoPrivate::networkModeDirectoryList(DeviceInfoPrivate::NetworkMode mode)
+{
+    if (!m_networkModeDirectoryListHash.contains(mode)) {
+        QStringList &modeDirectoryList(m_networkModeDirectoryListHash[mode]);
+        QDir baseDir(QStringLiteral("/sys/class/net"));
+        QStringList stemList;
+        if (mode == DeviceInfoPrivate::WlanMode)
+            stemList << QStringLiteral("wlan");
+        else if (mode == DeviceInfoPrivate::EthernetMode)
+            stemList << QStringLiteral("eth") << QStringLiteral("usb") << QStringLiteral("rndis");
+        for (auto stemIter = stemList.cbegin(); stemIter != stemList.cend(); ++stemIter) {
+            QFileInfoList modeDirList(baseDir.entryInfoList(QStringList() << QStringLiteral("%1*").arg(*stemIter), QDir::Dirs, QDir::Name));
+            for( auto modeDirIter = modeDirList.cbegin(); modeDirIter != modeDirList.cend(); ++modeDirIter)
```

Comment:
Nit: for()

Reply:
Fixed, along with one case of 'type& " -> 'type &"